app/controllers/duties_controller.rb

```diff
@@ -1,2 +1,58 @@
 class DutiesController < ApplicationController
+  def index
+    date = params[:date]
+    starting_date = date - (date.cwdays - 1)
+    @duties = Array.new(7)
+    starting_date.upto(starting_date + 7) do |day|
+      @duties.push(day)
+    end
+    @time_ranges = Time_range.all
+    @places = Place.all
+  end
+
+  def process_grab
+    @user_id = User.find(user_id_params)
+    @duty_id = Duty.find(duty_id_params)
+
+    # user_id of duty_id should and must be null
+    redirect_to action: 'index' if @duty_id.update(@user_id)
+  end
+
+  def process_drop
+    @user_id = User.find(user_id_params)
+    @duty_id = Duty.find(duty_id_params)
+
+    redirect_to action: 'index' if @duty_id.update(:user_id, nil)
+  end
+
+  def edit
```

Comment:
We're going to do a mass update, just like in the old duty website. Maybe check the current admin panel?